app/controllers/email_authorizations_controller.rb

```diff
@@ -0,0 +1,12 @@
+class EmailAuthorizationsController < ApplicationController
+  def verify
+    user = User.find_by(username: params[:username])
+    raise ActionController::RoutingError, "Not Found" unless user && current_user == user
+
+    email_authorization = user.email_authorizations.order("created_at DESC").first
+    raise ActionController::RoutingError, "Not Found" unless email_authorization.confirmation_token == params[:confirmation_token]
+
+    email_authorization.update(verified_at: Time.now.utc)
```

Comment:
`Time.current` as `Time.now` objects technically have no timezone, it's good to keep the convention